client/src/components/publications/form/ProfileLinkPage.js

```diff
@@ -18,9 +20,29 @@ const ProfileLinkPage = ({ closeModal }) => {
     profileLink: '',
   };
 
+  const validation = (values) => {
+    // extracting the authorId from the profileLink
+    const position = values.profileLink.indexOf('user=');
+    if (position === -1) {
+      dispatch({
+        type: IMPORT_FAIL,
+        payload: 'Please provide a valid profile link',
+      });
+      return false;
+    } else {
+      gScholarId = values.profileLink.substring(position + 5, position + 17);
+      dispatch({
+        type: UPDATE_GSCHOLAR_ID,
+        payload: gScholarId,
+      });
+      return true;
+    }
+  };
+
```

Comment:
Wasn't sure how else to pull up the validation from the actions, please let me know if it can be better. This was pulled up after discussion with @pmyip3 since the validation logic didn't need to be in the action.